characterisation/triggers/TRPS0002.test.ts

```diff
@@ -0,0 +1,52 @@
+import World from "../../utils/world"
+import generatePhase2Message from "../helpers/generatePhase2Message"
+import { processPhase2Message } from "../helpers/processMessage"
+import MessageType from "../types/MessageType"
+import { TriggerCode } from "../types/TriggerCode"
+
+const code = TriggerCode.TRPS0002
+
+describe.ifPhase2("TRPS0002", () => {
+  afterAll(async () => {
+    await new World({}).db.closeConnection()
+  })
+
+  it("creates a TRPS0002 for AnnotatedHearingOutcome when no operations and exceptions are generated and result code is 3107", async () => {
+    const inputMessage = generatePhase2Message({
+      messageType: MessageType.ANNOTATED_HEARING_OUTCOME,
+      hoTemplate: "NoOperationsAndExceptions",
+      offences: [{ results: [{ cjsResultCode: 3107 }] }]
+    })
+
+    const { triggers } = await processPhase2Message(inputMessage, { recordable: false })
+
+    expect(triggers).toContainEqual({ code })
```

Comment:
A bit of nitpick and preference thing, but I prefer if we hardcoded the trigger code to remove that bit of cognitive load. Personally don't mind the duplication.

```suggestion
    expect(triggers).toContainEqual({ code: TriggerCode.TRPS0002 })
```